packages/cli/src/UserManagement/PermissionChecker.ts

```diff
@@ -62,10 +62,14 @@ export class PermissionChecker {
 
 		const nodeToFlag = credIdsToNodes[inaccessibleCredIds[0]][0];
 
-		throw new NodeOperationError(nodeToFlag, 'Node has no access to credential', {
-			description: 'Please recreate the credential or ask its owner to share it with you.',
-			level: 'warning',
-		});
+		throw new NodeOperationError(
+			nodeToFlag,
+			`Node ${nodeToFlag.name} has no access to credential ID ${inaccessibleCredIds[0]} in workflow ${workflow.name} $(ID ${workflow.id})`,
```

Comment:
instead of adding all the data in the message/description, I'd much prefer if we serialize that data in the responses we send to the frontend, and then improve the handling of such errors in the frontend. We should really move away from using error messages as storage for metadata.

Reply:
Agree, but refactoring how we send/display might not be trivial and this is meant as a quick help for a customer. Realistically it might need to wait? Let me know otherwise.